--- engine/pathfinder/astar.cpp.gcc	2008-03-01 21:15:35.000000000 +0100
+++ engine/pathfinder/astar.cpp	2010-03-28 15:22:27.000000000 +0200
@@ -33,6 +33,8 @@
 --  Includes
 ----------------------------------------------------------------------------*/
 
+#include <cstdlib>
+#include <cstdio>
 #include "stratagus.h"
 #include "pathfinder.h"
 
--- engine/stratagus/stratagus.cpp.gcc	2008-03-01 21:15:36.000000000 +0100
+++ engine/stratagus/stratagus.cpp	2010-03-28 15:22:10.000000000 +0200
@@ -160,6 +160,7 @@
 #include <time.h>
 #include <ctype.h>
 #include <sstream>
+#include <algorithm>
 
 #ifndef _MSC_VER
 #include <unistd.h>
--- engine/stratagus/translate.cpp.gcc	2008-01-27 00:02:54.000000000 +0100
+++ engine/stratagus/translate.cpp	2010-03-28 15:22:10.000000000 +0200
@@ -36,6 +36,7 @@
 
 #include <map>
 #include <string>
+#include <cstdio>
 
 #include "translate.h"
 #include "iolib.h"
